tests/PHPStan/Analyser/data/preg-match-type-specifying.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace PregMatchTypeSpecifying;
+
+use function PHPStan\Testing\assertType;
+
+class Foo {
+	public function unknownFlags(int $flags)
+	{
+		// we don't know the flags, therefore allow all possible variations.
+		if (preg_match('/^[a-z]+$/', 'foo', $matches, $flags)) {
+			assertType('array<int<0, max>|string, array{string|null, int<-1, max>}>', $matches);
```

Comment:
Shouldn't it be this?

```suggestion
			assertType('array<int<0, max>|string, array{string|null, int<-1, max>}|string|null>', $matches);
```

It seems to default to PREG_CAPTURE_OFFSET if no flag is known but IMO it should probably default to `array<int<0, max>|string, string|null>` in the odd case no flag can be established, because PREG_CAPTURE_OFFSET is kinda the exception and adding an array as possible value there will most likely cause lots of false positives.